androidshared/src/main/java/org/odk/collect/androidshared/ui/DialogService.kt

```diff
@@ -0,0 +1,109 @@
+package org.odk.collect.androidshared.ui
+
+import android.content.Context
+import android.view.View
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+
+object DialogService {
```

Comment:
@grzesiek2010 I'd be interested to know how you feel about using our own machinery to fake AndroidX's `AlertDialog` as opposed to using a custom Shadow like I do in https://github.com/getodk/collect/pull/5391/commits/fa5116e3405cc1502eb04e8e9adb6f99c963dae3. There's definitely some more clean up on `DialogService` to be done, but I'd like to get input on the general approach before moving forward as I'm not feeling particularly confident about whether it's better than using Robolectric or not.

Reply:
The second approach looks much more natural to me but if you feel like that `our own machinery to fake AndroidX's AlertDialog` makes life/testing easier I won't be against it.

Reply:
Great, thanks! I can see advantages and disadvantages to both, so if you feel like using a custom Shadow is better, let's go with that.